### ISSUE TITLE
Add ability to comment on votes

### ISSUE THREAD

**Issue by DeveloppSoft** [aragon/aragonOS#91]
It would be nice to have the ability to comment on votes while selecting "yes" or "no", I think it could lead to some debates.
Storing the comment string wouldn't be necessary, an event should be sufficient.
Thoughts?

If the idea is accepted I can implement it.

**Comment by izqui** [comment]
Even if the string is not stored it has to be part of the tx payload and the event (leading to increased gas costs). Also it would limit commenting to only the moment you are voting and only once. 

I think commenting in the context of a vote is very interesting but I'm skeptical about it being onchain at all.

**Comment by DeveloppSoft** [comment]
Hm yes indeed, this should probably part of the UI then, maybe by storing comments on ipfs?

**Comment by evok3d** [comment]
Yeh i agree. I think it would be great to have the option to poll on a number of options rather than Yes vs No - as in some cases its not that simple.

Cheers

**Comment by lkngtn** [comment]
I think the importance of facilitating discussion/debate related to voting and decisions cannot be understated. Though I agree with the above comments that it doesn't necessarily need to be on-chain, or even directly connected to the vote. 

I do think there would be some advantage to having the discussions tied to the addresses/identities used for voting would be useful though. IE if you have a certain share and vote a certain way, you can leave a comment that justifies or argues your position. This would be particularly useful in the context of delegative voting.

**Comment by luisivan** [comment]
I agree with the importance of this feature, and I also agree that we should keep it offchain. The process could be that first the comment is signed by its creator, and then stored in IPFS. I think this also requires having some sort of organization-wide storage, which was planned for 0.5/0.6, and would make this much easier

**Comment by onbjerg** [comment]
We could look into [IPLD](https://github.com/ipld/ipld) for threading comments.

**Comment by DeveloppSoft** [comment]
If you guys want to keep things on the ethereum network, you could also look at `whisper`, if I am correct Status.im uses it for its messenger.

**Comment by onbjerg** [comment]
@DeveloppSoft The issue with Whisper is that currently the routing sucks, so messages might never reach the recipient. Status "solves" this by waiting for a read receipt, and if it no read receipt is sent, they will rebroadcast the message indefinitely.

Another issue is also that clients must be online to receive Whisper messages. It is therefore not possible to see what people have commented if you're not online 24/7.

**Comment by DeveloppSoft** [comment]
Oh ok didn't know that

**Comment by sohkai** [comment]
Moving this over to https://github.com/aragon/aragon-apps/issues/298, since that's where the apps are now :).
